lib/factory_bot_rails/lazy_registry_find.rb

```diff
@@ -0,0 +1,16 @@
+require "factory_bot/registry"
+
+module FactoryBotRails
+  module LazyRegistryFind
+    def self.find_definitions_once
+      return if defined?(@definitions_loaded)
```

Comment:
@quadule 👋 Fellow user of this gem... I like the concept of this PR... but, this concerns me, how does this interface with FactoryBot.reload? Does this caching get cleared on reload?

In other words, do we get a new registry when the configuration is reset in [reset_configuration](https://github.com/thoughtbot/factory_bot/blob/8446cb6c5b39ea046d8ba180197aabc66adf62ed/lib/factory_bot/reload.rb#L3)?